app-games/src/main/java/com/aptoide/android/aptoidegames/gamegenie/presentation/GameGenieUseCase.kt

```diff
@@ -1,55 +1,84 @@
 package com.aptoide.android.aptoidegames.gamegenie.presentation
 
+import cm.aptoide.pt.feature_apps.data.AppMapper
+import com.aptoide.android.aptoidegames.gamegenie.data.GameGenieAppRepository
 import com.aptoide.android.aptoidegames.gamegenie.data.database.model.GameGenieHistoryEntity
 import com.aptoide.android.aptoidegames.gamegenie.domain.ChatInteraction
+import com.aptoide.android.aptoidegames.gamegenie.domain.ChatInteractionHistory
 import com.aptoide.android.aptoidegames.gamegenie.domain.ConversationInfo
+import com.aptoide.android.aptoidegames.gamegenie.domain.GameContext
 import com.aptoide.android.aptoidegames.gamegenie.domain.GameGenieChat
+import com.aptoide.android.aptoidegames.gamegenie.domain.GameGenieChatHistory
 import com.aptoide.android.aptoidegames.gamegenie.domain.Token
+import com.aptoide.android.aptoidegames.gamegenie.io_models.GameGenieMetadata
 import com.aptoide.android.aptoidegames.gamegenie.io_models.GameGenieRequest
 import com.aptoide.android.aptoidegames.gamegenie.io_models.GameGenieResponse
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
 import kotlinx.coroutines.flow.map
 import javax.inject.Inject
 
 private const val MAX_CHATS = 15
 
 class GameGenieUseCase @Inject constructor(
   private val gameGenieManager: GameGenieManager,
+  private val mapper: AppMapper,
+  private val appRepository: GameGenieAppRepository,
 ) {
   suspend fun getToken(): Token {
     return gameGenieManager.getToken()
   }
 
+  fun getInstalledApps(): Flow<List<GameContext>> = appRepository.getInstalledApps()
+
   suspend fun reloadConversation(chat: GameGenieChat): GameGenieChat {
     val lastMessage = chat.conversation.lastOrNull()?.user ?: ""
     return if (lastMessage.isNotEmpty())
-      sendMessage(chat, lastMessage)
+      sendMessage(chat.toGameGenieChatHistory(), lastMessage, emptyList())
     else
       chat
   }
 
   suspend fun sendMessage(
-    chat: GameGenieChat,
+    chat: GameGenieChatHistory,
     userMessage: String,
+    installedApps: List<GameContext>,
   ): GameGenieChat {
     val updatedConversation = chat.conversation.toMutableList().apply {
       if (isNotEmpty()) {
         this[lastIndex] = last().copy(user = userMessage)
       }
     }
-    return postMessage(chat.id, chat.title, updatedConversation).fold(
+    return postMessage(chat.id, chat.title, updatedConversation, installedApps).fold(
```

Comment:
Said it before but just to pin point where it is being used, should we be passing the installed apps on every message or should we make like a sync when we open the view, for example ?